HikVisionSource/README.md

```diff
@@ -0,0 +1,217 @@
+this can be controlled by parameters outlined in the config section.
+The extension source can support mutiple cameras in parallel, based on configuration.  
+
+In order to incorporate this Extension Source, you will need to set up your local machine with a client JRE that can connect to your HikVision camera.
+Once you have done this, you will need to create the Source in Vantiq.
+The documentation
+has been split into two parts, [Setting Up Your Machine](#machine) and [Setting Up Your VANTIQ](#vantiq).
+
+# Prerequisites <a name="pre" id="pre"></a>
+
+In order to compile correctly, you must download the Hikvision SDK from [*here*](https://www.hikvision.com/en/support/download/sdk/), choosing the Device Network SDK (for Linux 64 bit).
+For windows based system, you should download the Device Network SDK (for Windoes 64 bit) as well from the same location.
+Make certain to use the correct implementation for the environment within which you are running.
+
+**IMPORTANT:** Read the [Testing](#testing) section before building this project.
+
+# Setting Up Your Machine <a name="machine" id="machine"></a>
+
+## Repository Contents
+
+# Setting Up Your Machine <a name="machine" id="machine"></a>
+
+## Repository Contents
+
+* **HikVisionMain** -- The main function for the program. Connects to sources as specified in the configuration file.
+* **HikVisionCore** -- Coordinates the connections to Vantiq, responsible for recovery the connection with Vantiq Server
+* **HikVisionHandleConfiguration** -- Sets up the source based on Vantiq configuration.
+* **HikVision** -- Thie class implememnts the SDK of HikVision Cameras. Supporting multiple cameras, allwoing publish commands and manages the camera notifications.
+* **HCNetSDK** -- The implementation of the HikVision jna interface.
+
+## Prerequisite
+
+For windows based systems, one should download the Device Network SDK (for Windows 64 bit) from [*here*](https://www.hikvision.com/en/support/download/sdk/), extract the lib folder, set it access using an environment parameter. 
+please refer to `How to run for distribution` section
+
+Define the environment variable **HIKVISION_LOC**.
+This should be the directory where the jna.jar and example.jar files are found.
+Those need to be extracted from the Device Network SDK (for Linux 64 bit) available as outlined above, under the folder LinuxJavaDemo direcory which included in the downloaded archive.
+
+**NOTE** the above jar files are relevant to windows based system as well. 
+
+The Linux version required the follwong files as well:
+
+* **libhcnetsdk.so**
+* **libhpr.so**
+* **libHCCore.so**
+* **libHCPreview.so** which is in the HCNetSDKCom directory of the lib path of the Device Network SDK (for Linux 64 bit)
+
+**NOTE**  For easier soluiton one might include all the content of the lib directory from Hikvision SDK to the lib directory of extension source, including the subdirectories HCNetSDKCom content in that lib directory
```

Comment:
I think what you really need to do here (and I think @olivier-bernard is looking at that) is to change HIKVISION_LOC to point to the parent directory of what you ask above.  Then, you can include things from HIKVISION_LOC/lib & LinuxJavaDemo (or whatever it's called).

```suggestion
**NOTE**  For an easier solution, one might include all the content of the lib directory from Hikvision SDK to the lib directory of extension source, including the subdirectories HCNetSDKCom content in that lib directory
```